src/data/polynomial/erase_lead.lean

```diff
@@ -266,4 +266,48 @@ lemma map_nat_degree_eq_nat_degree {S F : Type*} [semiring S]
   (φ p).nat_degree = p.nat_degree :=
 (map_nat_degree_eq_sub (λ f h, (nat.not_lt_zero _ h).elim) (by simpa)).trans p.nat_degree.sub_zero
 
+lemma card_support_eq_one : f.support.card = 1 ↔ ∃ (k : ℕ) (x : R) (hx : x ≠ 0), f = C x * X ^ k :=
```

Comment:
Is it easier to have an n-ary version something like
```suggestion
lemma card_support_eq : f.support.card = n
  ↔ ∃ (k : fin n -> ℕ) (x : fin n  -> R) (hk : monotone k) (hx : \forall i, x i ≠ 0), f = \sum i, C (x i) * X ^ (k i) :=
```
and then prove the versions for small `n` in terms of it?

Reply:
I tried this approach for a bit, and it seems doable, but more difficult.

Reply:
I think the n-ary version should still be there even if it doesn't give shorter proofs of the specific cases.

Reply:
Yes, it just that it's rather painful to prove.

Reply:
To extend function why not using `[function.extend](https://leanprover-community.github.io/mathlib_docs/logic/function/basic.html#function.extend)`?

Reply:
In any case I agree we want both versions, if the more general one is a pain it can b done in another PR.